appservice/src/deploy/runWithZipStream.ts

```diff
@@ -80,17 +69,23 @@ const commonGlobSettings: {} = {
 /**
  * Adds files using glob filtering
  */
-function addFilesGlob(zipper: archiver.Archiver, folderPath: string): void {
+export async function getFilesFromGlob(folderPath: string): Promise<string[]> {
     const zipDeployConfig: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration(ext.prefix, vscode.Uri.file(folderPath));
     const globPattern: string = zipDeployConfig.get<string>('zipGlobPattern') || '**/*';
-    const ignorePattern: string | string[] | undefined = zipDeployConfig.get<string | string[]>('zipIgnorePattern');
-    zipper.glob(globPattern, { cwd: folderPath, ignore: ignorePattern, ...commonGlobSettings });
+    const filesToInclude: string[] = await globby(globPattern, { cwd: folderPath });
+
+    const ignorePattern: string | string[] = zipDeployConfig.get<string | string[]>('zipIgnorePattern') || '';
+    const filesToIgnore: string[] = await globby(ignorePattern, { cwd: folderPath });
+
+    return filesToInclude.filter(file => {
+        return !filesToIgnore.includes(file);
+    })
 }
 
 /**
  * Adds files using gitignore filtering
  */
-async function addFilesGitignore(zipper: archiver.Archiver, folderPath: string, gitignoreName: string): Promise<void> {
+async function getFilesFromGitignore(folderPath: string, gitignoreName: string): Promise<string[]> {
```

Comment:
Looks like `globby` has a `gitignore` method, too. Might be worth switching over to that instead of `glob-gitignore`, but I'm not sure if that should be in this PR.

Reply:
What were your thoughts on this? Part of me wants to just move over to be consistent while we're changing things up, but I also understand keeping changes minimal at this point has value, too

Reply:
I'd rather not mess with it right now.  Since we'd theoretically be releasing this with Functions and this purely affects Functions, I'd rather wait until after release since this is more an engineering thing (assuming functionality is the same)